weave/trace/tests/test_op_return_forms.py

```diff
@@ -86,3 +87,528 @@ async def fn():
     assert res.calls[0].op_name == obj_ref.uri()
     assert res.calls[0].inputs == {}
     assert res.calls[0].output == 1
+
+
+def simple_list_accumulator(acc, value):
+    acc.append(value)
+    return acc
+
+
+def test_op_return_sync_generator(client):
```

Comment:
these are a permutations of tests: async vs non-async, generator vs iterator, none vs partial vs error vs full = 16 tests